x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheUtils.java

```diff
@@ -20,4 +30,56 @@ public class BlobCacheUtils {
     public static int toIntBytes(long l) {
         return ByteSizeUnit.BYTES.toIntBytes(l);
     }
+
+    public static void throwEOF(long channelPos, long len, Object file) throws EOFException {
```

Comment:
Let's add a unit test to ensure it effectively throws in case someone changes that behavior in the future

Reply:
How do you mean? Just a test that passes random inputs and verifies that it always throws? Why would anybody change that behavior?:) It's the only line in the method?

Reply:
Oh sorry I meant to have some basic test coverage on `readSafe()` that throws using this method. We have many `Streams.read*` and now `BlobCacheUtils.readSafe()` utility methods but we are lacking basic test coverage I think